src/promptflow-azure/promptflow/azure/_utils/_artifact_client.py

```diff
@@ -68,11 +70,9 @@ async def register_artifact(self, run_id, datastore_name, relative_path, path):
                     raise UserAuthenticationError(error_message)
                 elif response.status_code != 200:
```

Comment:
Nit: shall we only retry retriable status codes like `http_retry_wrapper`?

Reply:
Saw that implementation, feel like it's based on a strong assumption about server side behavior with which I doubt if we are really familiar, especially considering we are dealing with metric/artifact/asset service, so I would prefer to retry for any internal error to be a safe way.

Reply:
Additionally, the metric server error I have met is that there is no status code at all

Reply:
I just concern retrying 500s may not work. e.g. if service has a bug returned 500, retry won't resolve the problem.